test/k8sT/Services.go

```diff
@@ -412,6 +412,93 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 	})
 
+	Context("Checks local redirect policy", func() {
+		var (
+			deploymentYAML string
+			lrpServiceName string
+			app1IP         string
+			app2IP         string
+			be1IP          string
+			be2IP          string
+		)
+
+		BeforeAll(func() {
+			deploymentYAML = helpers.ManifestGet(kubectl.BasePath(), "lrp-test.yaml")
+			lrpServiceName = "lrp-demo-service"
+			res := kubectl.ApplyDefault(deploymentYAML)
+			res.ExpectSuccess("Unable to apply %s", deploymentYAML)
+			waitPodsDs()
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(deploymentYAML)
+		})
+
+		getSingletonPodIP := func(label string) string {
+			podsIPs, err := kubectl.GetPodsIPs(helpers.DefaultNamespace, label)
+			Expect(err).To(BeNil(), "Cannot get pod IP for %s pod", label)
+			Expect(len(podsIPs)).To(Equal(1), "Got more than 1 ip for %s pod", label)
+			for _, ip := range podsIPs {
+				return ip
+			}
+			// Should never reach here
+			return ""
+		}
+
+		It("LRP connectivity", func() {
+			// Basic sanity check
+			ciliumPods, err := kubectl.GetCiliumPods()
+			Expect(err).To(BeNil(), "Cannot get cilium pods")
+			for _, pod := range ciliumPods {
+				service := kubectl.CiliumExecMustSucceed(context.TODO(), pod, "cilium service list", "Cannot retrieve services on cilium pod")
```

Comment:
Shall we grep for `clusterIP` to make sure that that service is of the LR type?

Reply:
This is after LRP has been deployed, hence the service type should be of string "LocalRedirect", are you suggesting we check service type *before* deploying LRP too?

Reply:
I meant to grep for a particular service to see whether it's type has changed to LocalRedirect, i.e. `fmt.Sprintf("cilium service list | grep %s", clusterIP)`.

Reply:
Ah I see, done.